src/main/java/com/puppycrawl/tools/checkstyle/Main.java

```diff
@@ -341,6 +341,29 @@ else if (hasSuppressionLineColumnNumber) {
         return result;
     }
 
+    /**
+     * Calls the {@link TreeStringPrinter} class's method with the provided file.
+     *
+     * @param file The file to parse.
+     * @param clss The custom implementation of {@link TreeStringPrinter} to use for printing.
+     * @throws IOException Failed to open a file
+     * @throws CheckstyleException error while parsing the file
+     */
+    private static void callTreeStringPrinter(File file, Class<?> clss)
+            throws IOException, CheckstyleException {
+        final String output;
+        try {
+            output = ((TreeStringPrinter) clss.getDeclaredConstructor().newInstance())
+                    .printFileAst(file);
+        }
+        catch (NoSuchMethodException | InvocationTargetException | InstantiationException
```

Comment:
There should be an entry for `ClassCastException`, otherwise no readable error will be shown to the user:
```bash
java -jar checkstyle.jar -pc com.puppycrawl.tools.checkstyle.Main

Exception in thread "main" java.lang.ClassCastException: class com.puppycrawl.tools.checkstyle.Main cannot be cast to class com.puppycrawl.tools.checkstyle.api.TreeStringPrinter (com.puppycrawl.tools.checkstyle.Main and com.puppycrawl.tools.checkstyle.api.TreeStringPrinter are in unnamed module of loader 'app')
	at com.puppycrawl.tools.checkstyle.Main.callTreeStringPrinter(Main.java:356)
	at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:304)
	at com.puppycrawl.tools.checkstyle.Main.execute(Main.java:197)
	at com.puppycrawl.tools.checkstyle.Main.main(Main.java:132)

Reply:
Suggested change:
```java
- catch (NoSuchMethodException | InvocationTargetException | InstantiationException
-                 | IllegalAccessException ex) {
+ catch (ClassCastException | ReflectiveOperationException ex) {
```

Reply:
Done in the next push.